### PR TITLE
[Feature] Mechas breakthrough

### PR DESCRIPTION
**Pull request by DeityLink**
- Mechas sent from Mass Drivers now break through windows, grilles, tables, and hurt people

https://dl.dropboxusercontent.com/u/33846352/byond/dreamseeker%202015-05-10%2005-38-31-07.webm

### CODE REVIEW

**Comment by FudgePucker** [comment]
>HoS sends centcom a fax saying that they all stink like doo doo.
Two deathsquad mechas bust though the glass windows.
"I HERD U TALKIN SHIT LIK I WOULDN'T HEAR."

Merge it now.

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -323,8 +325,63 @@
...
+			src.throwing = 0//so mechas don't get stuck when landing after being sent by a Mass Driver
+			src.crashing = null
+
+		if(breakthrough)
+			if(crashing)
+				spawn(1)
+					src.throw_at(crashing, 50, src.throw_speed)
+			else
+				spawn(1)
+					crashing = get_step(loc,dir)//is there a better way to get a turf 3 tiles away in a given direction?)
```
Comment (RemieRichards) [inline code/game/mecha/mecha.dm]: just use ```get_edge_target_turf()```
It's what is usually used for "just throw it in this direction"

Comment (DeityLink) [inline code/game/mecha/mecha.dm]: Yes but get_edge_target_turf() will send them all the way to the other side won't it? I want the mechs to stop exactly 3 tiles after they hit the first obstacle.

Comment (RemieRichards) [inline code/game/mecha/mecha.dm]: They'd slide (because throw_at()) a little bit like all atoms, but would of course slow down.

Comment (DeityLink) [inline code/game/mecha/mecha.dm]: they won't slide past the turf you give them in throw_at(), no matter the speed.

**Comment by DeityLink** [comment]
@RemieRichards There, I made a new proc, much cleaner like this.

**Comment by RemieRichards** [comment]
Yep, that's good.

File: code/game/mecha/mecha.dm [hunk 2]
```diff
@@ -323,8 +325,61 @@
...
+
+		else if(istype(obstacle, /obj/structure/reagent_dispensers/fueltank))
+			obstacle.ex_act(1)
+
+		else if(istype(obstacle, /mob/living))
+			var/mob/living/L = obstacle
+			var/hit_sound = list('sound/weapons/genhit1.ogg','sound/weapons/genhit2.ogg','sound/weapons/genhit3.ogg')
+			if(L.flags & INVULNERABLE)
+				return
+			L.take_overall_damage(5,0)
```
Comment (d3athrow) [inline code/game/mecha/mecha.dm]: >get rammed by a mecha going fast enough to bust through shit
>only 5 damage

**Comment by DeityLink** [comment]
aw yiss

**Comment by FudgePucker** [comment]
Does this mean deathsquids will be done more?

**Comment by DeityLink** [comment]
Sadly not until I can do my Strike Team revamp(which I won't do as long as there is even ONE map PR openned)

**Comment by FudgePucker** [comment]
Poop.
Why don't we just assign a number to every map PR, roll a dice, if it comes up that number, it gets merged, then everyone has to put up the map PR and it gets merged really fast or rejected and have to go to the end of the line if it is shit.

**Comment by DeityLink** [comment]
still a better idea that what we currently have.

**Pull request merged**